nerdery-ecommerce/src/auth/guards/access-token-with-roles.guard.ts

```diff
@@ -0,0 +1,65 @@
+import {
+    CanActivate,
+    ExecutionContext,
+    Injectable,
+    ForbiddenException,
+    ContextType,
+    UnauthorizedException,
+    BadGatewayException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { JwtPayloadDto } from '../dto/jwtPayload.dto';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { AccessTokenGuard } from './access-token.guard';
+import { ROLES_KEY } from '../decoratos/roles.decorator';
+
+@Injectable()
+export class AccessTokenWithRolesGuard extends AccessTokenGuard {
+    constructor(private readonly reflector: Reflector) {
+        super();
+    }
+
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+
+
+        const requiredRole = this.reflector.get<string>(
+            ROLES_KEY,
+            context.getHandler(),
+        );
+
+        console.log('requiredRole', requiredRole);
+        if (!requiredRole) {
+            return true;
+        }
+
+        const isAuthenticated = await super.canActivate(context);
```

Comment:
First auth authentication and the check roles!
1st Unauthorized,
2nd Forbbiden exceptions